frontend/app/views/support/Dates.scala

```diff
@@ -59,18 +59,31 @@ object Dates {
 
   case class Range(start: String, end: String)
 
-  def dateRange(start: DateTime, end: DateTime): Range = {
-    def dateToks(dt: DateTime) = dt.toString("EEEE d MMMMM YYYY").split(" ").reverse
+  private def buildDateRange(start: DateTime, end: DateTime, fmt: String): Range = {
```

Comment:
The code is pretty scary to look at, not surprised you thought it weird. The tests give some indication of what's going on: https://github.com/guardian/membership-frontend/blob/ebf425fab2/frontend/test/views/support/DatesTest.scala#L7-L42

The code is trying to display a date range (start-end) in a human-friendly way that is pleasantly concise by not unnecessarily repeating any parts of the string. Eg. do not do: 

```
Thursday 6 November 2014, 10.20am - Thursday 6 November 2014, 12.30pm
Thursday 6 November 2014 - Saturday 8 November 2014
```

...when you can say:

```
Thursday 6 November 2014, 10.20am - 12.30pm
Thursday 6 - Saturday 8 November 2014
```

If the start & end are on the same day, we display the date with the start-end times, otherwise we omit the time and use the second form, which removes duplication by truncating the right-hand-side of the 'start' string by all the parts that are already in the 'end' string.

I hope that explains it a little, though I'm not sure if it immediately suggests a better name.